src/Rules/Library/NameIsReasonableLength.cs

```diff
@@ -26,8 +26,8 @@ public NameIsReasonableLength()
 
         public override EvaluationCode Evaluate(IA11yElement e)
         {
-            if (e == null) throw new ArgumentException(nameof(e));
-            if (e.Name == null) throw new ArgumentException(nameof(e.Name));
+            if (e == null) throw new ArgumentNullException(nameof(e));
+            if (e == null) throw new ArgumentNullException(nameof(e));
```

Comment:
Lines 29 and 30 are identical. I suspect that line 30 is supposed to be:
```suggestion
            if (e.name == null) throw new ArgumentException(ErrorMessages.ElementNameNullOrWhiteSpace, nameof(e));
```